### ISSUE TITLE
Incorrect warnings when importing getter and setter from different libraries

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#25624]
Attempting to import a getter named `x` and setter named `x=` from different libraries produces the following analyzer warning at the location where the setter is used: `'x' cannot be used as a setter, it is final`. 

This warning is incorrect.  There's nothing in the spec requiring a getter and its corresponding setter to be imported from the same library; the two entities have different names so they should not conflict with each other.

Dart2js and the VM also mishandle this situation, but not in the same way.

I will submit shared tests shortly demonstrating the bug.

**Comment by bwilkerson** [comment]
This is probably caused by the fact that we create synthetic variables for getters and setters. They're useful for being able to associate the two, but they cause other problems. We should consider getting rid of them.

**Comment by stereotype441** [comment]
@bwilkerson Agreed, getting rid of the synthetic variables would probably fix this bug, and would probably clean things up elsewhere by bringing analyzer's behavior more in line with the spec.
